test/integration/scheduler/taint_test.go

```diff
@@ -0,0 +1,317 @@
+	}
+
+	// Case 2: No MemoryPressure Toleration for BestEffort pod.
+	besteffortPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "best-effort-pod",
+			Namespace: nsName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "busybox",
+					Image: "busybox",
+				},
+			},
+		},
+	}
+
+	besteffortPodInServ, err := clientset.CoreV1().Pods(nsName).Create(besteffortPod)
+	if err != nil {
+		t.Errorf("Case 2: Failed to create pod: %v", err)
+	} else if len(besteffortPodInServ.Spec.Tolerations) != 0 {
+		t.Errorf("Case 2: Unexpected toleration # of BestEffort pod, expected: 0, got: %v",
+			len(besteffortPodInServ.Spec.Tolerations))
+	}
+
+	// Case 3: Taint Node by NetworkUnavailable condition.
+	networkUnavailableNode := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "node-1",
+		},
+		Status: v1.NodeStatus{
+			Capacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("4000m"),
+				v1.ResourceMemory: resource.MustParse("16Gi"),
+				v1.ResourcePods:   resource.MustParse("110"),
+			},
+			Allocatable: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("4000m"),
+				v1.ResourceMemory: resource.MustParse("16Gi"),
+				v1.ResourcePods:   resource.MustParse("110"),
+			},
+			Conditions: []v1.NodeCondition{
+				{
+					Type:   v1.NodeNetworkUnavailable,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1.NodeReady,
+					Status: v1.ConditionFalse,
+				},
+			},
+		},
+	}
+
+	if _, err := clientset.CoreV1().Nodes().Create(networkUnavailableNode); err != nil {
+		t.Errorf("Case 3: Failed to create node: %v", err)
+	} else {
+		nodeInformer := informers.Core().V1().Nodes().Informer()
+		informerCh := make(chan bool)
+
+		nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(old, cur interface{}) {
+				curNode := cur.(*v1.Node)
+				for _, taint := range curNode.Spec.Taints {
+					if taint.Key == algorithm.TaintNodeNetworkUnavailable &&
+						taint.Effect == v1.TaintEffectNoSchedule {
+						informerCh <- true
+						break
+					}
+				}
+			},
+		})
+		select {
+		case <-time.After(60 * time.Second):
+			t.Errorf("Case 3: Failed to taint node after 60s.")
+		case <-informerCh:
+		}
+	}
+
+	// Case 4: Schedule Pod with NetworkUnavailable Taint.
+	networkDaemonPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "network-daemon-pod",
+			Namespace: nsName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "busybox",
+					Image: "busybox",
+				},
+			},
+			Tolerations: []v1.Toleration{
+				{
+					Key:      algorithm.TaintNodeNetworkUnavailable,
+					Operator: v1.TolerationOpExists,
+					Effect:   v1.TaintEffectNoSchedule,
+				},
+			},
+		},
+	}
+
+	if _, err := clientset.CoreV1().Pods(nsName).Create(networkDaemonPod); err != nil {
+		t.Errorf("Case 4: Failed to create pod for network daemon: %v", err)
+	} else {
+		podInformer := informers.Core().V1().Pods().Informer()
```

Comment:
Similarly, we may want to add the event handler before creating the pod.

Reply:
done